### ISSUE TITLE
Manual Save sometimes required 

### ISSUE THREAD

**Issue by olzama** [delph-in/matrix#558]
The questionnaire won't always successfully save the data form. In some cases, manually clicking "save" is required:

1) in order to see the newly-entered values in relevant drop-down boxes;
2) in order to serialize the choices.

Example: the Other Features page. Enter a FORM feature name. It should then appear as an available supertype, but it does not, unless "Save" is clicked.

<img width="584" alt="Screen Shot 2021-01-12 at 10 45 54 AM" src="https://user-images.githubusercontent.com/10963114/104358844-d8590280-54c3-11eb-86c4-6bf0b3bb26bb.png">

**Comment by emilymbender** [comment]
I've never been quite sure when & how the saving happens, actually, so this may be a long-standing bug. Even just a clear sense of when it happens would be helpful. (E.g. does navigating to a different subpage trigger a 'save'?)

**Comment by olzama** [comment]
Yeah.

> E.g. does navigating to a different subpage trigger a 'save'?

Sometimes :) :) ;(

**Comment by olzama** [comment]
...which would probably indicate that is has to do something with which page we are navigating *from*.

**Comment by olzama** [comment]
Possibly related:

When adding a new FORM value, an exception occurs, which may prevent autosaving. 

<img width="759" alt="Screen Shot 2021-01-12 at 12 38 42 PM" src="https://user-images.githubusercontent.com/10963114/104370110-3ccf8e00-54d3-11eb-84be-f2ee1fbd481c.png">

When saved manually, the added value appears as normal.

**Comment by olzama** [comment]
Looks like "forms" may not be a great variable name for javascript......

**Comment by olzama** [comment]
I don't think it has to do with using "form" for the variable name after all. The same exception occurs e.g. on the Person subpage, in that at certain points, javascript cannot find an element with the ID person-subtype2-button. Form, like person, uses the "-subtype" prefix; maybe that's the problem? Except I did try to get rid of it, for Form, and it did not help.

**Comment by olzama** [comment]
Hmm in fact, if in the case with FORM, saving helps, in case with numbers, I can't get it to work at all... This surely hasn't been like this before? I wonder when it started.
